x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/indexlifecycle/action/TransportExplainLifecycleAction.java

```diff
@@ -65,19 +76,32 @@ protected void doMasterOperation(ExplainLifecycleRequest request, String[] concr
             IndexMetaData idxMetadata = state.metaData().index(index);
             Settings idxSettings = idxMetadata.getSettings();
             String policyName = LifecycleSettings.LIFECYCLE_NAME_SETTING.get(idxSettings);
+            String currentPhase = LifecycleSettings.LIFECYCLE_PHASE_SETTING.get(idxSettings);
+            // parse existing phase steps from the phase definition in the index settings
+            String phaseDef = idxSettings.get(LifecycleSettings.LIFECYCLE_PHASE_DEFINITION);
+            PhaseExecutionInfo phaseExecutionInfo = null;
+            if (Strings.isNullOrEmpty(phaseDef) == false) {
+                try (XContentParser parser = JsonXContent.jsonXContent.createParser(xContentRegistry,
+                        DeprecationHandler.THROW_UNSUPPORTED_OPERATION, phaseDef)) {
+                    phaseExecutionInfo = PhaseExecutionInfo.parse(parser, currentPhase);
+                } catch (IOException e) {
+                    logger.error("failed to parse [" + LifecycleSettings.LIFECYCLE_PHASE_DEFINITION + "] for index [" + index + "]", e);
```

Comment:
I don't think we should swallow this error here, I think the request should error out also